### ISSUE TITLE
Fedora 34: org.telegram.desktop does not start: AL lib: (EE) Error enumerating devices: Capture Stream did not get ready (Timeout)

### ISSUE THREAD

**Issue by sanjayankur31** [flathub/org.telegram.desktop#242]
Hi there,

I noticed that the Telegram Flatpak stopped working for me recently. I'm on Fedora 34 though, so that could have something to do with it. Here's what I see:

```
$ flatpak run --user org.telegram.desktop 

** (process:2): WARNING **: 12:32:00.343: Unfortunately, GTK integration conflicts with qgtk2 platformtheme and style. Therefore, QT_QPA_PLATFORMTHEME and QT_STYLE_OVERRIDE will be unset.
** Message: 12:32:00.344: This can be ignored by setting DESKTOP_APP_I_KNOW_ABOUT_GTK_INCOMPATIBILITY environment variable to any value, however, if qgtk2 theme or style is used, this will lead to a crash.
** Message: 12:32:00.344: GTK integration can be disabled by setting DESKTOP_APP_DISABLE_GTK_INTEGRATION to any value. Keep in mind that this will lead to some features being unavailable.
QSocketNotifier: Can only be used with threads started with QThread
AL lib: (EE) Error enumerating devices: Capture Stream did not get ready (Timeout)
```

Info on the installation:

```
$ flatpak info --user org.telegram.desktop 

Telegram Desktop - Telegram Desktop messenger

          ID: org.telegram.desktop
         Ref: app/org.telegram.desktop/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 2.6.1
     License: GPL-3.0
      Origin: flathub
  Collection: org.flathub.Stable
Installation: user
   Installed: 67.3 MB
     Runtime: org.kde.Platform/x86_64/5.15
         Sdk: org.kde.Sdk/x86_64/5.15

      Commit: 2cb02b4037d74e2e8d6a8e5af6931a0dd56d31cff2d28b42d894332ecefe2138
      Parent: 2fa46b503435b6f97f5045cd4919a2c6e4ba952cfe3b5cb1ccc20ae8483ed63e
     Subject: Update tdesktop.git to 2.6.1 (a00067ef)
        Date: 2021-02-24 21:48:02 +0000

$ rpm -qa \*flatpak\*
flatpak-libs-1.10.2-1.fc34.x86_64
flatpak-session-helper-1.10.2-1.fc34.x86_64
flatpak-selinux-1.10.2-1.fc34.noarch
flatpak-1.10.2-1.fc34.x86_64
```

Please do let me know if there's anything I can help to diagnose the issue.

**Comment by ilya-fedin** [comment]
Probably there's something wrong with your audio stack?

**Comment by gasinvein** [comment]
This might have something to do with Fedora's pipewire switch. Try running via `flatpak run --filesystem=xdg-run/pipewire-0 org.telegram.desktop`

**Comment by ilya-fedin** [comment]
Shouldn't pipewire-pulse handle this gracefully? Personally I'm using NixOS with pipewire 0.3.21 and don't have any issues with tdesktop from flatpak.

**Comment by gasinvein** [comment]
Probably it should, but I know very little of how it works and interacts with flatpak. Just seen discussions mentioning that pipewire-by-default in upcoming Fedora does affect flatpak apps.

**Comment by ilya-fedin** [comment]
AFAIK, it emulates pulse socket, by the same path. So no need in extra dependencies. Probably there are some bugs in the pulse shim that @sanjayankur31 should report to pipewire.

**Comment by sanjayankur31** [comment]
I've tried the usual steps:

- downgrade pipewire and reboot,
- create a brand new user and install telegram etc

No luck either way, I'm afraid.

I've not tinkered with my audio setup at all, and it seems to work fine in the browser etc. I'm happy to report the issue to pipewire. Would you folks have any pointers on what information to collect? @gasinvein : would you have a link to the discussion? I may be able to chime in there or get some ideas on what to try?

Thanks for your help.

**Comment by ilya-fedin** [comment]
> Would you folks have any pointers on what information to collect?

Probably you should find you pipewire version first and check that it's the latest for upstream. If yes, then you can go directly to pipewire upstream and file a bug there. If not, then you should go to your distro bugtracker and file a bug there.
You can get detailed openal log with `ALSOFT_LOGLEVEL=3` variable.

**Comment by gasinvein** [comment]
> No luck either way, I'm afraid.

Did `--filesystem=xdg-run/pipewire-0` not help either?

> would you have a link to the discussion? I may be able to chime in there or get some ideas on what to try?

That mostly happened on `#flatpak` IRC channel on freenode, I'm not sure if it has public logs (probably it does, though).

@nanonyme knows more about pipewire and it's interaction with flatpak runtime, probably this issue rings some bells.

**Comment by ilya-fedin** [comment]
> Did `--filesystem=xdg-run/pipewire-0` not help either?

It won't do anything. OpenAL talks with $XDG_RUNTIME_DIR/pulse/native socket.

**Comment by ilya-fedin** [comment]
> That mostly happened on `#flatpak` IRC channel on freenode, I'm not sure if it has public logs (probably it does, though).

I don't think it's related to flatpak sandbox, this seems to be a bug in pulse emulating. I had some similar errors at the time of pipewire 0.3.17.

**Comment by karuboniru** [comment]
an workaround:
run `pactl load-module module-null-sink object.linger=1 media.class=Audio/Source/Virtual sink_name=my-source channel_map=front-left,front-right` before starting telegram

See: https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/624

**Comment by sanjayankur31** [comment]
Ah, that worked! Thanks very much @karuboniru .

(I do not have a mic on this system, but it hadn't ever caused trouble before. )

**Comment by ilya-fedin** [comment]
Looks like pipewire released 0.3.24, closing

**Comment by NotAFile** [comment]
Just installed fedora 34 and it looks like this still occurs. The pactl workaround works for me too.

**Comment by ilya-fedin** [comment]
Maybe bugged fedora 34 release has a bugged pipewire version?

**Comment by sanjayankur31** [comment]
@NotAFile : have you run a system update yet?  There's a freeze in place before the release, so there are quite a few 0 day updates which fix issues that were not considered blockers by the QA team.

This is what my up to date F34 has now:

```
$ rpm -qa \*pipewire\*
pipewire0.2-libs-0.2.7-5.fc34.x86_64
pipewire-libs-0.3.26-3.fc34.x86_64
pipewire-0.3.26-3.fc34.x86_64
pipewire-gstreamer-0.3.26-3.fc34.x86_64
pipewire-alsa-0.3.26-3.fc34.x86_64
pipewire-jack-audio-connection-kit-0.3.26-3.fc34.x86_64
pipewire-pulseaudio-0.3.26-3.fc34.x86_64
pipewire-utils-0.3.26-3.fc34.x86_64
```

**Comment by ilya-fedin** [comment]
@sanjayankur31 you have .26, so you don't have the issue anymore, yeah?

**Comment by sanjayankur31** [comment]
I haven't experienced it lately, but I also haven't rebooted in a while so I can't currently be sure that the workaround is not still in place.

**Comment by hfiguiere** [comment]
restarting pipewire `systemctl --user restart pipewire` often helps

**Comment by NotAFile** [comment]
@sanjayankur31 
> @NotAFile : have you run a system update yet? There's a freeze in place before the release, so there are quite a few 0 day updates which fix issues that were not considered blockers by the QA team.

Yep, I am on 0.3.26 too. I'll have to try restarting pipewire next time. But it sounds like this bug is unrelated to tdesktop so I might take it up with them instead.

**Comment by ilya-fedin** [comment]
Would be really funny if they didn't fix the issue in the reality and people said it's fixed just because they applied the workaround (that is probably persistent even across restarts)

**Comment by sanjayankur31** [comment]
> Would be really funny if they didn't fix the issue in the reality and people said it's fixed just because they applied the workaround (that is probably persistent even across restarts)

I think this is the case. After a reboot, I hit the same error and had to use the workaround. 

Does anyone know if there are some pipewire confs or something that may need clearing up when updating to new pipewire versions?

**Comment by ilya-fedin** [comment]
> See: https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/624

btw they still haven't reopened the issue, probably someone should ask them clearly for that

**Comment by ilya-fedin** [comment]
So the developer reopened the issue 3 days ago, wrote something about dummy sources, I asked him why it hangs even without dummy source and he ignored me, seem to be a dead end :roll_eyes:

**Comment by ilya-fedin** [comment]
Reopening the issue since looks like this won't be solved any time soon

**Comment by ilya-fedin** [comment]
Can we close the issue?

**Comment by NotAFile** [comment]
I have received a few pw updates and not experienced this again.

**Comment by sanjayankur31** [comment]
```
$ rpm -qa \*pipewire\*
pipewire0.2-libs-0.2.7-5.fc34.x86_64
pipewire-0.3.30-2.fc34.x86_64
pipewire-libs-0.3.30-2.fc34.x86_64
pipewire-gstreamer-0.3.30-2.fc34.x86_64
pipewire-jack-audio-connection-kit-0.3.30-2.fc34.x86_64
pipewire-alsa-0.3.30-2.fc34.x86_64
pipewire-pulseaudio-0.3.30-2.fc34.x86_64
pipewire-utils-0.3.30-2.fc34.x86_64
```

Telegram works fine after reboot now, without using the workaround. Closing.
